### PR TITLE
Terrain estimator

### PR DESCRIPTION
**Pull request by tumbili**
Simple terrain estimator which can be combined very easily with any estimator.

### CODE REVIEW

**Comment by tumbili** [comment]
Simple terrain estimator which can be combined very easily with any estimator.

**Comment by mhkabir** [comment]
@tumbili Thanks for this! 

@jgoppert @AndreasAntener This ties in nicely with our LPE work. This will allow proper AGL handling and correction of local position from the terrain estimate.

**Comment by jgoppert** [comment]
@tumbili Looking forward to trying this out!

**Comment by mhkabir** [comment]
@tumbili I have a general suggestion - could you make the velocity update a general thing instead of directly using GPS? The estimator would just send it's current velocity estimate. We need this for the GPS-denied situation (e.g when people are using vision localization or a flow-only setup)

**Comment by tumbili** [comment]
@mhkabir Sure, I can do that, makes sense.
@jgoppert The reason why I've written the estimator was to test fixed wing auto landings with the ekf. I found it very hard to follow the ekf logic so I used this standalone estimator. I'm very interested to try out your position estimator soon. Looks very clean and understandable, I like that :-)
